videoprocess/vacopy.cpp

```diff
@@ -0,0 +1,853 @@
+        return -1;
+    }
+
+    std::cout<<"=========Media Copy========="<<endl;
+
+    if (g_src.memtype == VA_SURFACE_ATTRIB_MEM_TYPE_VA)
+    {
+        std::cout<<"copy from 2D tile surface to ";
+    }
+    else
+    {
+        if (g_src.alignsize == 1 || !(g_src.width%g_src.alignsize))
+            std::cout<<"copy from 1D linear surface to ";
+        else
+            std::cout<<"copy from 2D linear surface with pitch_align "<<g_src.alignsize<<" to ";
+    }
+
+    if (g_dst.memtype == VA_SURFACE_ATTRIB_MEM_TYPE_VA)
+    {
+        std::cout<<"2D tile surface."<<endl;
+    }
+    else
+    {
+        if (g_dst.alignsize == 1 || !(g_dst.width%g_dst.alignsize))
+            std::cout<<"1D linear surface."<<endl;
+        else
+            std::cout<<"2D linear surface with pitch_align "<<g_dst.alignsize<<endl;
+    }
+    std::cout << "prefer hw engine is "<<g_copy_method<<". notification, 0: blanance(vebox), 1: perf(EU), 2 powersaving(blt)"<<endl;
+
+    return 0;
+}
+
+static void
+print_help()
+{
+    printf("The app is used to test the scaling and csc feature.\n");
+    printf("Cmd Usage: ./vacopy process_copy.cfg\n");
+    printf("The configure file process_copy.cfg is used to configure the para.\n");
+    printf("You can refer process_copy.cfg.template for each para meaning and create the configure file.\n");
+}
+int32_t main(int32_t argc, char *argv[])
+{
+    VAStatus va_status;
+    uint32_t i;
+
+    if (argc != 2 || !strcmp(argv[1], "-h") || !strcmp(argv[1], "--help")){
+        print_help();
+        return -1;
+    }
+
+    /* Parse the configure file for video process*/
+    strncpy(g_config_file_name, argv[1], MAX_LEN);
+    g_config_file_name[MAX_LEN - 1] = '\0';
+
+    if (NULL == (g_config_file_fd = fopen(g_config_file_name, "r"))){
+        printf("Open configure file %s failed!\n",g_config_file_name);
+        assert(0);
+    }
+
+    /* Parse basic parameters */
+    if (parse_basic_parameters()){
+        printf("Parse parameters in configure file error\n");
+        assert(0);
+    }
+
+    va_status = vpp_context_create();
+    if (va_status != VA_STATUS_SUCCESS) {
+        printf("vpp context create failed \n");
+        assert(0);
+    }
+
+    /* Video frame fetch, process and store */
+    if (NULL == (g_src.fd = fopen(g_src.name, "r"))){
+        printf("Open SRC_FILE_NAME: %s failed, please specify it in config file: %s !\n",
+                g_src.name, g_config_file_name);
+        assert(0);
+    }
+
+    if (NULL == (g_dst.fd = fopen(g_dst.name, "w"))){
+        printf("Open DST_FILE_NAME: %s failed, please specify it in config file: %s !\n",
+               g_dst.name, g_config_file_name);
+        assert(0);
+    }
+
+    printf("\nStart to process, ...\n");
+    struct timespec Pre_time;
+    struct timespec Cur_time;
+    unsigned int duration = 0;
+    clock_gettime(CLOCK_MONOTONIC, &Pre_time);
+
+    for (i = 0; i < g_frame_count; i ++){
+        upload_frame_to_surface(g_src.fd, g_in_surface_id);
```

Comment:
The clock calculation should just include the vaCopy time consumption. The video data upload and store surface to file is not needed.